### PR TITLE
feat: add support for the issuedOnBehalfOf X.509v3 extension

### PR DESCRIPTION
**Pull request by JonathanWilbur**
This PR adds support for the `issuedOnBehalfOf` X.509v3 extension defined in ITU-T Recommendation X.509. It's syntax is defined as:

```asn
issuedOnBehalfOf EXTENSION ::= {
  SYNTAX         GeneralName
  IDENTIFIED BY  id-ce-issuedOnBehalfOf }
```

##### Checklist
- [x] tests are added or updated

### CODE REVIEW

**Review by Sashan — CHANGES_REQUESTED** [review]
Thank you looks good to me. I have just minor/bikeshedding suggestion on `i2r_IOBO()`.

File: crypto/x509/v3_iobo.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/x509v3.h>
+#include "ext_dat.h"
+
+static int i2r_IOBO(X509V3_EXT_METHOD *method,
```
[the remark this answers is not in the document]
Comment (JonathanWilbur) [reply crypto/x509/v3_iobo.c]: I fixed this. I also fixed the copyright year at the top of the file!

**Review by Sashan — APPROVED** [review]
Thank you for changing the function name.
all looks good to me.

**Review by nhorman — APPROVED** [review]
just the one copyright nit

File: crypto/x509/v3_iobo.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 1999-2024 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (t8m) [inline crypto/x509/v3_iobo.c:2]: This should be AFAIK just 2024

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Assuming @nhorman approval holds.

**Comment by nhorman** [comment]
Ack, approval still good

**Review by shahsb — APPROVED** [review]
LGTM.

Do we need to add Documentation changes for this?

**Comment by JonathanWilbur** [comment]
I fixed the merge conflict resulting from `auditIdentity` getting merged in.

**Comment by JonathanWilbur** [comment]
@t8m or @mattcaswell can I get a merge or re-review, if needed? This one has been `done` for four days.

**Comment by t8m** [comment]
As this was a trivial rebase, I do not think this needs re-approvals.
Merging.

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**